Add plugin reference info to plugin entity doc (#312)
Kong/developer.konghq.com: app/_gateway_entities/plugin.md

* Plugin Development Kit reference
* Other Language Support

## Scoping plugins

You can run plugins in various contexts, depending on your environment needs.
Each plugin can run globally, or be scoped to some combination of the following:
* [Gateway Services](/gateway/entities/service/)
* [Routes](/gateway/entities/route/)
* [Consumers](/gateway/entities/consumer/)
* [Consumer Groups](/gateway/entities/consumer-group/)

Using scopes, you can customize how {{site.base_gateway}} handles functions in your environment, 
either before a request is sent to your backend services or after it receives a response.
For example, if you apply a plugin to a single [**Route**](/gateway/entities/route/), 
that plugin will only trigger when a request matches the Route's specific path.

### Global scope

> A global plugin is not associated to any Service, Route, Consumer, or Consumer Group is 

Reviewer: ```suggestion
A global plugin is not associated with any Service, Route, Consumer, or Consumer Group. It is 
```
I took my best guess at what it was supposed to be saying after "Consumer Group". I think it was supposed to be two different sentences.

considered global, and will be run on every request, regardless of any other configuration.

* In self-managed {{site.ee_product_name}}, the plugin applies to every entity in a given Workspace.

> * In self-managed {{site.ce_product_name}}, the plugin applies to your entire environment.

Reviewer: {{site.ce_product_name}} needs to be added as a variable. Currently, it's blank on the rendered preview.

* In {{site.konnect_short_name}}, the plugin applies to every entity in a given Control Plane.

Every plugin supports a subset of these scopes.

### Plugin precedence

A single plugin instance always runs _once_ per request. The
configuration with which it runs depends on the entities it has been
configured for.
Plugins can be configured for various entities, combinations of entities, or even globally.
This is useful, for example, when you want to configure a plugin a certain way for most requests but make _authenticated requests_ behave slightly differently.

Therefore, there is an order of precedence for running a plugin when it has been applied to different entities with different configurations. The number of entities configured to a specific plugin directly correlates to its priority. The more entities a plugin is configured for, the higher its order of precedence.

The complete order of precedence for plugins configured to multiple entities is:

1. **Consumer** + **Route** + **Service**: Highest precedence, affecting authenticated requests that match a specific Consumer on a particular Route and Service.

> 2. **Consumer group** + **Service** + **Route**: Affects groups of authenticated users across specific Services and Routes.

Reviewer: ```suggestion
2. **Consumer Group** + **Service** + **Route**: Affects groups of authenticated users across specific Services and Routes.
```

3. **Consumer** + **Route**: Targets authenticated requests from a specific Consumer on a particular Route.
4. **Consumer** + **Service**: Applies to authenticated requests from a specific Consumer accessing any Route within a given Service.

> 5. **Consumer group** + **Route**: Affects groups of authenticated users on specific Routes.

Reviewer: ```suggestion
5. **Consumer Group** + **Route**: Affects groups of authenticated users on specific Routes.
```

> 6. **Consumer group** + **Service**: Applies to all Routes within a specific Service for groups of authenticated users.

Reviewer: ```suggestion
6. **Consumer Group** + **Service**: Applies to all Routes within a specific Service for groups of authenticated users.
```

7. **Route** + **Service**: Targets all Consumers on a specific Route and Service.
8. **Consumer**: Applies to all requests from a specific, authenticated Consumer across all Routes and Services.
9. **Consumer Group**: Affects all Routes and Services for a designated group of authenticated users.

> 10. **Route**: Specific to given Route.

Reviewer: ```suggestion
10. **Route**: Affects the given Route.
```

> 11. **Service**: Specific to given Service. 

Reviewer: ```suggestion
11. **Service**: Affects the given Service. 
```

12. **Globally configured plugins**: Lowest precedence, applies to all requests across all Services and Routes regardless of Consumer status.

{:.info}
> **Note on precedence for Consumer Groups**:
When a Consumer is a member of two Consumer Groups, each with a scoped plugin, {{site.base_gateway}} ensures deterministic behavior by executing only one of these plugins. Currently, this is determined by the Group name, in alphabetical order. 

> However, the specific rules that govern this behavior are not defined and are subject to change in future releases.

Reviewer: Clarify this.

Does this mean that if a Consumer is a member of Group A and Group B and the Rate Limiting Advanced plugin is scoped to group A and separately to group B, it will only run one if the plugins (Rate Limiting Advanced) are the same? 

I'd think if the plugins were different (JWT is scoped to Group A and Rate Limiting is scoped to Group B), it would run both plugins since they do different things/are different plugins.
